metaflow/plugins/cards/component_serializer.py

```diff
@@ -357,7 +716,50 @@ def extend(self, components):
 
             return
 
-        self._cards_components[self._default_editable_card].extend(components)
+        self._card_component_store[self._default_editable_card].extend(components)
+
+    @property
+    def components(self):
+        # FIXME: document
+        if self._default_editable_card is None:
+            if len(self._card_component_store) == 1:
+                card_type = list(self._cards_meta.values())[0]["type"]
+                _warning_msg = [
+                    "Card of type `%s` is not an editable card." % card_type,
+                    "Components list will not be updated and `current.card.components` will not work for any call during this runtime execution.",
+                    "Please use an editable card",  # todo : link to documentation
+                ]
+            else:
+                _warning_msg = [
+                    "`current.card.components` cannot disambiguate between multiple @card decorators.",
+                    "Components list will not be accessible and `current.card.components` will not work for any call during this runtime execution.",
+                    "To fix this set the `id` argument in all @card when using multiple @card decorators over a single @step and reference `current.card[ID].components`",  # todo : Add Link to documentation
+                    "to update/access the appropriate card component.",
+                ]
+            if not self._warned_once["components"]:
+                self._warning(" ".join(_warning_msg))
+                self._warned_once["components"] = True
+            return
+
+        return self._card_component_store[self._default_editable_card].components
```

Comment:
@tuulos : Something interesting is happening in this case where we _may_ need to change the behavior. The current behavior of cards is that `current.card.append` will result in warnings when you try to add components to a card that doesn't exist or when `current.card` cannot figure out the default editable card. 

Now that we are supporting runtime updates, we need to have a clear story of how we deal with this scenario: 

```python
# `current.card` cannot disambiguate the default editable card
current.card.append(componentA, id="a") # this will throw a warning. 

current.card.components["a"].update(...) # This will crash since currently we are not returning anything
componentA.update(...) # this will work, but nothing will change on the UI
```

The main point is that if `append` and `extend` only throw warnings, it will also affect how the code behaves when `update` is called. Should we now throw exceptions when users try to `append` to cards that cannot be disambiguated?

Reply:
the main case we want to handle is card updates hidden deep inside a library, referring to a card that you forgot to add `--with card`. To continue the policy that cards are best-effort and will never crash the user code, we could handle even
```
current.card.components["a"].update(...)
```
leniently and make a non-existing ID return a stub object that just shows a warning. I don't think this is a common case but being lenient sometimes and not other times is confusing too. We could just be always lenient - just show warnings, never crash